granulate_utils/java.py

```diff
@@ -215,3 +217,94 @@ def parse_jvm_version(version_string: str) -> JvmVersion:
         zing_major = None
 
     return JvmVersion(version, build, vm_name, vm_type, zing_major)
+
+
+@dataclasses.dataclass
+class JvmFlag:
+    name: str
+    type: str
+    value: str
+    origin: str
+    kind: List[str]
+
+    vm_flags_pattern = re.compile(
+        r"(?P<flag_type>\S+)\s+"
+        r"(?P<flag_name>\S+)\s+"
+        r"(?P<flag_equal_sign_prefix>:)?= "
+        r"(?P<flag_value>\S+)\s+"  # noqa: E501 # We don't support empty string nor spaces in flag values, although both are legal values
+        r"{(?P<flag_kind>.+?)}"
+        r"(?:\s*{(?P<flag_origin_jdk_9>.*)})?"
+    )
+
+    def to_dict(self) -> Dict[str, Union[str, List[str]]]:
+        return dataclasses.asdict(self)
+
+    @classmethod
+    def from_dict(cls, jvm_flag_dict: Dict[str, Any]) -> JvmFlag:
+        return cls(**jvm_flag_dict)
```

Comment:
Consider removing the method and just using `JvmFlag(...)` anywhere you use `JvmFlag.from_dict(...)`.

Reply:
I prefer to leave it and `to_dict` for ease of use and clarity